CARTAvis/carta-frontend
# Yuhsuan/1852 check new release

**Description**

Closes #1852: show a notification when there is a new CARTA release.

Implemented:
* When the frontend is initialized (with preferences updated), check the release tags on GitHub.
(ref: https://docs.github.com/en/rest/releases/releases#list-releases)
* If the latest release tag is not the same as the latest release version stored in the preference (default: `"v" + version info` in package.json), "send a message" to the users:
  
  <img width="194" alt="image" src="https://user-images.githubusercontent.com/43841102/181721016-d4c3ea8d-87e0-430c-af7d-3343f3f2c7f7.png">  <img width="215" alt="image" src="https://user-images.githubusercontent.com/43841102/181720930-33e12fac-3647-4437-ba22-dbb997ea3c34.png">

* Click the message button to show the notification:
  beta release
  <img width="423" alt="image" src="https://user-images.githubusercontent.com/43841102/181722422-3f045776-4f01-4be4-92e3-e84c9f43d9a8.png">
  stable release
  <img width="422" alt="image" src="https://user-images.githubusercontent.com/43841102/181722364-731ac5e0-b7e2-4f92-975d-266d032233b5.png">
* Click the message button again or else where to close the notification. The message will remain at the corner.
* Click `OK` button to dismiss the message. The message will not be shown again until another new release is available.
* If `don't show` toggle is on when clicking the `OK` button, the message will never be shown again.
* Manually reset this feature by running `app.preferenceStore.clearPreferences(["checkNewRelease", "latestRelease"]);` in the code snippet or the console.
* Adjusted the alignment of top right corner indicators
  <img width="147" alt="image" src="https://user-images.githubusercontent.com/43841102/181717578-92bb460f-6329-45f4-943d-f4002850482b.png">  <img width="143" alt="image" src="https://user-images.githubusercontent.com/43841102/181717639-4247528b-b25b-40c4-aa3c-974804529523.png">

How to test it:
* Manually set the compared version by `app.preferenceStore.setPreference("latestRelease", "xxx");`
* Simulate a Github release update by `app.updateNewRelease("xxx");`

@veggiesaurus @ajm-asiaa Please check if the version comparison works, thanks.

~ToDo: we may want to skip this feature in some situations, e.g. read-only mode?~
The preference still works with local storage in a read-only mode, so we can keep this feature.

**Checklist**
- [ ] e2e test passing / added corresponding fix
